test/Benchmarks/Metrics/ExemplarBenchmarks.cs

```diff
@@ -0,0 +1,93 @@
+// <copyright file="ExemplarBenchmarks.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using BenchmarkDotNet.Attributes;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+
+/*
+// * Summary *
+BenchmarkDotNet=v0.13.3, OS=Windows 11 (10.0.22621.1265)
+11th Gen Intel Core i7-1185G7 3.00GHz, 1 CPU, 8 logical and 4 physical cores
+.NET SDK=7.0.103
+  [Host]     : .NET 7.0.3 (7.0.323.6910), X64 RyuJIT AVX2
+  DefaultJob : .NET 7.0.3 (7.0.323.6910), X64 RyuJIT AVX2
+
+
+|          Method | EnableExemplar |     Mean |   Error |  StdDev | Allocated |
+|---------------- |--------------- |---------:|--------:|--------:|----------:|
+| HistogramUpdate |          False | 249.6 ns | 4.42 ns | 7.13 ns |         - |
+| HistogramUpdate |           True | 284.2 ns | 5.39 ns | 5.04 ns |         - |
```

Comment:
not bad 👋

Reply:
this does not use the spec prescribed API for Reservoir. THe spec mandated one does not allow passing the histogram index, so it'd require finding the histogram index *twice* (this is how java implemented it now).

Will try that and share numbers as well for ref.